lib/ex_sdp/origin.ex

```diff
@@ -41,34 +40,39 @@ defmodule ExSDP.Origin do
           username: binary(),
           session_id: integer(),
           session_version: integer(),
-          address: ConnectionData.sdp_address()
+          address: Address.t()
         ) :: t()
   def new(opts \\ []) do
```

Comment:
Haven't noticed that earlier, but origin in `ExSDP.new` can now default to `Origin.new`. Moreover, `ExSDP.new` could accept `username`, `session_id` etc instead of `origin`, so that the user didn't have to do `ExSDP.new(ExSDP.Origin.new(...`

Reply:
There are other params/attributes that we create in this way i.e. we at first create struct and then pass it to some function e.g. `ConnectionData` `Timing` or `RtpMapping` so maybe leaving this as it is will be more consistent?

Reply:
Some SDP lines, including `origin`, occur exactly once in SDP, so their parameters could be placed in SDP struct itself. Though, since having a struct for each SDP line is pretty intuitive, I thought to leave them and simplify the `new` function, but I don't insist. If needed, SDP can be constructed from scratch, and with the use of all the structs.